pkg/common/common.go

```diff
@@ -145,3 +146,93 @@ func IsContextDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+// get a list of IP addresses configured on the host's
+// interfaces.
+func GetLocalIPs() ([]string, error) {
+	var localIPs []string
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return []string{}, err
```

Comment:
@rchincha, do you happen to have any thoughts on how we could test for these error scenarios where we're querying something from the system?

Reply:
https://github.com/project-zot/zot/tree/main/pkg/test/inject
It is for these kinds of scenarios, we added this facility. Take a look, however, the downside is code becomes somewhat unreadable for an uninformed reader.